src/Compilers/Core/Portable/Operations/Queries/IQueryContinuation.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Semantics
+{
+    /// <summary>
+    /// Represents a query continuation in C#.
+    /// </summary>
+    /// <remarks>
+    /// This interface is reserved for implementation by its associated APIs. We reserve the right to
+    /// change it in the future.
+    /// </remarks>
+    public interface IQueryContinuation : IOperation
```

Comment:
>IQueryContinuation [](start = 21, length = 18)

Should we treat IQueryContinuation as just another query clause? Then we might be able to strongly type QueryBody as IQueryClause and probably can do the same for IQueryExpression.LastClauseOrContinuation.